calculator.rb

```diff
@@ -0,0 +1,203 @@
+# Jacquelyn Cheng
+# Ada Cohort 10 Nodes
+# Calculator
+# https://github.com/Ada-C10/calculator
+# 8/7/2018
+
+=begin
+
+Primary Requirements:
+The program should use the input operation and two numbers to provide the
+result of applying the operation to the two numbers.
+√   The program should have support for these four operations: addition,
+    subtraction, multiplication, and division.
+√   The program should accept both the name (add) and the symbol (+) for each
+    possible operation.
+√   The program should handle unexpected user input
+
+Optional Enhancements:
+√   Print out the formula in addition to the result, i.e. 2 + 2 = 4
+√   Add support for computing exponents (2^4 = 2 * 2 * 2 * 2 = 16).
+√   Add support for the modulo operator (10 % 3 = 1).
+√   Gracefully handle unexpected user input:
+        What happens if the user input is nil (i.e., the user pressed enter)?
+        What happens if the user tries to add hotdog to elephant?
+√   Make your program know when it needs to return an integer versus a float.
+X   Add support for parentheticals, i.e. 10 / (5 + 5) = 1.
+
+Notes on further improvements to be made:
+-   Would be great to be able to accept fractions too
+
+=end
+
+# flow of program:
+#   welcome
+#   ask for operators
+#   validate operator
+#   ask for number1
+#   validate number1
+#   ask for number 2
+#   validate number 2
+#   calculate formula
+#   calculate answer
+#   display formula and answer
+
+
+# validate if the user input is a symbol or description of an operator
+# return the corresponding operator or nil if invalid
+
+def validate_operator_input(input)
+  case input
+    when "+", "add"
+      return "+"
+    when "-", "subtract"
+      return "-"
+    when "*", "multiply"
+      return "*"
+    when "/", "divide"
+      return "/"
+    when "^", "exponent"
+      return "**"
+    when "%", "remainder"
+      return "%"
+    else
+      # method to ask for operator input
+      return nil
+    end
```

Comment:
You didn't line up this `end` with the corresponding `case`, but everywhere else you did.